mbed_host_tests/host_tests_plugins/module_power_cycle_mbed.py

```diff
@@ -0,0 +1,179 @@
+"""
+mbed SDK
+Copyright (c) 2011-2015 ARM Limited
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Author: Przemyslaw Wirkus <Przemyslaw.Wirkus@arm.com>
+"""
+
+import os
+import json
+import time
+import requests
+from host_test_plugins import HostTestPluginBase
+
+
+class HostTestPluginPowerCycleResetMethod(HostTestPluginBase):
+
+    # Plugin interface
+    name = 'HostTestPluginPowerCycleResetMethod'
+    type = 'ResetMethod'
+    stable = True
+    capabilities = ['power_cycle']
+    required_parameters = ['target_id', 'device_info']
+
+    def setup(self, *args, **kwargs):
+        """! Configure plugin, this function should be called before plugin execute() method is used.
+        """
+        return True
+
+    def execute(self, capability, *args, **kwargs):
+        """! Executes capability by name
+
+        @param capability Capability name
+        @param args Additional arguments
+        @param kwargs Additional arguments
+        @details Each capability e.g. may directly just call some command line program or execute building pythonic function
+        @return Capability call return value
+        """
+        if 'target_id' not in kwargs or not kwargs['target_id']:
+            self.print_plugin_error("Error: This plugin requires mbed target_id")
+            return False
+
+        if 'device_info' not in kwargs or type(kwargs['device_info']) is not dict:
+            self.print_plugin_error("Error: This plugin requires dict parameter 'device_info' passed by the caller.")
+            return False
+
+        result = False
+        if self.check_parameters(capability, *args, **kwargs) is True:
+            if capability in HostTestPluginPowerCycleResetMethod.capabilities:
+                target_id = kwargs['target_id']
+                device_info = kwargs['device_info']
+                result = self.__hw_reset(target_id, device_info)
+        return result
+
+    def __hw_reset(self, target_id, device_info):
+        """
+        Performs hardware reset of target ned device.
+
+        :return:
+        """
+        result = False
+        try:
```

Comment:
You probably wanna check if these env variables are set before you actually allow people to reset? 
It is maybe not that important but it would be cool if ```__hw_reset()``` is not depending on these env.variables. Rather they are passed as parameters from ```execute()```.
Rationale: It would be easier to use ```__hw_reset()``` as API function so I can call it with IP and port and do not rely on env. settings.